zerver/lib/message.py

```diff
@@ -1642,3 +1642,10 @@ def update_to_dict_cache(
 
     cache_set_many(items_for_remote_cache)
     return message_ids
+
+
+def get_user_message_for_safe_bulk_update() -> QuerySet:
```

Comment:
This should be annotated as returning a `QuerySet[UserMessage]`.

Also, I think `get_user_message_select_for_update_queryset` would be a clearer name.